Upgrade `@types/node`

Amazing package, thank you for all of y'all's hard work!

I recently installed in a new project and got a dependency error on `@types/node`-- It appears you only allow versions `^16.11.6`, but some of my packages (vite, solidjs, etc.) require later versions. Is there any reason you can't upgrade your version? Or allow higher versions?

Honestly, I'm unsure of the significance of this (not super knowledgeable on package publishing), but I was able to `--force` my way through the install anyway, so I don't think it's a huge deal.

<details>
<summary>Show error log</summary>

```bash
npm error code ERESOLVE
npm error ERESOLVE unable to resolve dependency tree
npm error
npm error While resolving: obsidian-dataedit@0.0.0
npm error Found: @types/node@22.0.3
npm error node_modules/@types/node
npm error   dev @types/node@"^22.0.3" from the root project
npm error
npm error Could not resolve dependency:
npm error peer @types/node@"^16.11.6" from obsidian-typings@2.0.0
npm error node_modules/obsidian-typings
npm error   dev obsidian-typings@"*" from the root project
npm error
npm error Fix the upstream dependency conflict, or retry
npm error this command with --force or --legacy-peer-deps
npm error to accept an incorrect (and potentially broken) dependency resolution
```
</details>

--- a/package.json
+++ b/package.json
@@ -43,12 +43,13 @@
     },
     "homepage": "https://github.com/Fevol/obsidian-typings#readme",
     "peerDependencies": {
-        "@types/node": "^16.11.6",
-        "electron": "^30.0.9",
+        "@types/node": ">=14.0.0",
+        "electron": ">=1.6.10",
         "obsidian": "^1.6.6"
     },
     "devDependencies": {
         "@tsconfig/strictest": "^2.0.5",
+        "@types/node": "^18.17.0 || >=20.1.0",
         "builtin-modules": "^4.0.0",
         "cspell": "^8.13.0",
         "dprint": "^0.45.1",

